paasta_tools/chronos_tools.py

```diff
@@ -956,3 +961,147 @@ def is_temporary_job(job):
     :returns: a boolean indicating if a job is a temporary job
     """
     return job['name'].startswith(TMP_JOB_IDENTIFIER)
+
+
+ENTERED, ENTERED_AND_EXITED = range(0, 2)
+
+
+def dfs(node, get_adjacent_nodes, add_visited, node_status=None, ignore_cycles=False, callback_on_node_enter=None):
```

Comment:
can we make this function as close to referentially transparent as we can? I had a hard time reasoning about how it worked in ```_get_related_jobs_and_configs```.